core/prisma/seed/seedCommunity.ts

```diff
@@ -0,0 +1,1102 @@
+import type { Static } from "@sinclair/typebox";
+import type {
+	componentConfigSchemas,
+	componentsBySchema,
+	InputTypeForCoreSchemaType,
+} from "schemas";
+
+import { faker } from "@faker-js/faker";
+import { jsonArrayFrom } from "kysely/helpers/postgres";
+
+import type {
+	ActionInstancesId,
+	Communities,
+	CommunitiesId,
+	FormAccessType,
+	FormElements,
+	Forms,
+	FormsId,
+	NewMembers,
+	PubFields,
+	PubsId,
+	PubTypes,
+	Stages,
+	StagesId,
+	Users,
+	UsersId,
+} from "db/public";
+import {
+	Action as ActionName,
+	CoreSchemaType,
+	ElementType,
+	InputComponent,
+	MemberRole,
+	StructuralFormElement,
+} from "db/public";
+import { logger } from "logger";
+import { expect } from "utils";
+
+import type { actions } from "~/actions/api";
+import { db } from "~/kysely/database";
+import { createPasswordHash } from "~/lib/auth/password";
+import { createPubRecursiveNew } from "~/lib/server";
+import { slugifyString } from "~/lib/string";
+
+export type PubFieldsInitializer = Record<
+	string,
+	{
+		relation?: true;
+		schemaName: CoreSchemaType;
+	}
+>;
+
+type PubTypeInitializer<PF extends PubFieldsInitializer> = Record<
+	string,
+	Partial<Record<keyof PF, true>>
```

Comment:
I don't really understand why this is `<Record<keyof PF, true>>`? what is the `true` for? I noticed the 

```
"Navigation Item": {
  Title: true,
  "External Link": true,
  "Open In New Tab": true,
  // nested navigation
  "Navigation Items": true,
},
```
in the seed script, is the `true` significant?

Reply:
Good question! No, it's not really significant, this just defines which fields belong to which pubtype.
This could have also been an array I guess, but I find in general that it's easier for typescript to do things with Objects. I also left it like this because maybe in the future we would add some configuration to a PubField in a PubType, for instance marking a PubField as the "title" field. This would not be that hard to add in this way.